src/Akka.Hosting/Configuration/HoconConfigurationRoot.cs

```diff
@@ -0,0 +1,74 @@
+﻿// -----------------------------------------------------------------------
+//   <copyright file="HoconConfigurationRoot.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.EnvironmentVariables;
+using Microsoft.Extensions.Primitives;
+
+namespace Akka.Hosting.Configuration;
+
+internal class HoconConfigurationRoot: IConfigurationRoot
```

Comment:
This is just a wrapper over the official `IConfigurationRoot` implementation, it replaces any `EnvironmentVariablesConfigurationProvider` with our own implementation.